source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/teamexplorer/internal/TeamExplorerHelpers.java

```diff
@@ -279,4 +311,219 @@ public static IViewPart showView(final String viewID) {
+                    found = false;
+                    continue;
+                }
+
+                boolean repoUrlAvailable = false;
+                final String[] queryItems = uri.getQuery().split("&"); //$NON-NLS-1$
+
+                for (final String queryItem : queryItems) {
+                    if (StringUtil.startsWithIgnoreCase(queryItem, PROTOCOL_HANDLER_URL_PARAM)) {
+                        final String value = queryItem.substring(PROTOCOL_HANDLER_URL_PARAM.length());
+                        System.setProperty(PROTOCOL_HANDLER_URL_PROPERTY, value);
+                        repoUrlAvailable = tryParseGitRepoUrl(value);
+                    } else if (StringUtil.startsWithIgnoreCase(queryItem, PROTOCOL_HANDLER_ENCODING_PARAM)) {
+                        final String value = queryItem.substring(PROTOCOL_HANDLER_ENCODING_PARAM.length());
+                        System.setProperty(PROTOCOL_HANDLER_ENCODING_PROPERTY, value);
+                    } else if (StringUtil.startsWithIgnoreCase(queryItem, PROTOCOL_HANDLER_BRANCH_PARAM)) {
+                        final String value = queryItem.substring(PROTOCOL_HANDLER_BRANCH_PARAM.length());
+                        System.setProperty(PROTOCOL_HANDLER_BRANCH_PROPERTY, value);
+                    }
+                }
+
+                if (repoUrlAvailable) {
+                    return true;
+                } else {
+                    found = false;
+                }
+            } else if (arg.equalsIgnoreCase(PROTOCOL_HANDLER_ARG)) {
+                found = true;
+            }
+        }
+
+        return false;
+    }
+
+    public static boolean hasProtocolHandlerRequest() {
+        final String url = System.getProperty(PROTOCOL_HANDLER_URL_PROPERTY, null);
+        return !StringUtil.isNullOrEmpty(url);
+    }
+
+    public static String getProtocolHandlerServer() {
+        return System.getProperty(PROTOCOL_HANDLER_SERVER_PROPERTY, StringUtil.EMPTY);
+    }
+
+    public static String getProtocolHandlerProject() {
+        return System.getProperty(PROTOCOL_HANDLER_PROJECT_PROPERTY, StringUtil.EMPTY);
+    }
+
+    public static String getProtocolHandlerBranch() {
+        return System.getProperty(PROTOCOL_HANDLER_BRANCH_PROPERTY, StringUtil.EMPTY);
+    }
+
+    public static String getProtocolHandlerRepository() {
+        return System.getProperty(PROTOCOL_HANDLER_REPOSITORY_PROPERTY, StringUtil.EMPTY);
+    }
+
+    public static String getProtocolHandlerUrl() {
+        return System.getProperty(PROTOCOL_HANDLER_URL_PROPERTY, StringUtil.EMPTY);
+    }
+
+    public static String getProtocolHandlerEncodedUrl() {
+        return URIUtils.newURI(System.getProperty(PROTOCOL_HANDLER_URL_PROPERTY, StringUtil.EMPTY)).toASCIIString();
+    }
+
+    private static boolean tryParseGitRepoUrl(final String repoUrl) {
+        final URI uri = URIUtils.newURI(repoUrl);
+        final String[] pathItems = uri.getPath().split("/"); //$NON-NLS-1$
```

Comment:
I am not sure how this split method handles trailing separators (i.e. what happens is the url ends in a /)